resource_booking_combination_auto_create/tests/test_resource_combination_wizard.py

```diff
@@ -301,6 +302,69 @@ def test_select_resource_in_multiple_selected_categories(self):
         others.
         """
 
+    def test_no_dates(self):
+        """
+        Opening the wizard if the start date or the duration is not set should
+        raise a validation error.
+        """
+        booking = self.env["resource.booking"].create(
+            {
+                "partner_id": self.partner_1.id,
+                "type_id": self.booking_type_1.id,
+                # no start and no duration
+                # "start": "2023-02-06 10:00:00",
+                # "duration": 2,
+            }
+        )
+        with self.assertRaises(ValidationError) as cm:
+            self._create_wizard_with_selected_categories(
+                booking, [self.room_category, self.worker_category]
+            )
+        self.assertEqual(
+            str(cm.exception),
+            "To select resources, the booking must have a start date and a "
+            "duration.",
+        )
```

Comment:
Did cm survive the context? Shouldn't this be indented to the right?

Reply:
this seems very weird indeed, but that code would not be reachable if it would be indented to the right, as the context is exited as soon as the exception is raised. do you see another way of checking the message of the exception?

Reply:
from a little searching on the internet; apparently this is completely normal. see also other example.

```python
🐚 coopiteasy reuse-tool main $ with open("README.md") as fp:
...............................     text = fp.read()
...............................
🐚 coopiteasy reuse-tool main $ fp
<_io.TextIOWrapper name='README.md' mode='r' encoding='UTF-8'>
🐚 coopiteasy reuse-tool main $ fp.closed
True
```